python/ml4ir/applications/ranking/tests/test_ml4ir_sanity.py

```diff
@@ -0,0 +1,258 @@
+    """
+    separable = False
+    seed = args.seed
+    rng = check_random_state(args.seed)
+    best_accuracy = 0
+    best_accuracy_data = None
+    trials = 1000
+    while not separable and trials > 0:
+        samples = datasets.make_classification(n_classes=args.n_classes, n_samples=args.n_samples, n_features=args.n_features,
+                                               n_redundant=args.n_redundant, n_informative=args.n_informative,
+                                               n_clusters_per_class=args.n_clusters_per_class, flip_y=args.flip_y,
+                                               class_sep=args.class_sep, random_state=rng)
+        p = Perceptron(fit_intercept=True, max_iter=1000, verbose=0, random_state=seed, validation_fraction=0.0001)
+        p.fit(samples[0], samples[1])
+        acc = p.score(samples[0], samples[1])
+        if acc > best_accuracy:
+            best_accuracy = acc
+            best_accuracy_data = samples
+        acc = p.score(samples[0], samples[1])
+        if acc == 1.0:
+            break
+        trials -= 1
+    return best_accuracy_data[0], best_accuracy_data[1]
+
+
+def calculate_MRR(prediction, y):
+    """
+    Calculates the MRR given the predicted scores (prediction) and the actual clicks (y).
+    """
+    if sum(y) < 1:
+        return 0
+    pred = prediction
+    clickedDocIdxs = [i for i, x in enumerate(y) if x == 1.0]
+    MRR = 0
+    sortedPred = sorted(pred)[::-1]
+    for clickedDocIdx in clickedDocIdxs:
+        clickedDocScore = pred[clickedDocIdx]
+        # ties are resolved with the worst case (i.e. the correct doc will have the lowest place among other docs with same score)
+        equalScoreCount = sortedPred.count(clickedDocScore)
+        predictedRank = sortedPred.index(clickedDocScore) + equalScoreCount - 1.0
+        MRR += 1.0 / (predictedRank + 1.0)
+    MRR = MRR / len(clickedDocIdxs)
+    return MRR
+
+
+def check_MRR(df, model, featureCount=2):
+    """
+    calculated MRR for a dataset given a model.
+    """
+    MRR = 0
+    queries = df.groupby('query_id')
+    for qid, group in queries:
+        X = [group['f' + str(i)] for i in range(featureCount)]
+        pred = model.decision_function(np.array(X).T)
+        mrr = calculate_MRR(pred, group['clicked'])
+        MRR += mrr
+    return MRR/len(queries)
+
+
+def check_accuracy(df, model, featureCount=2):
+    X = [df['f'+str(i)] for i in range(featureCount)]
+    Y = df['clicked']
+    acc = model.score(np.array(X).T, Y)
+    return acc
+
+
+def train_sklearn_model(df, model, featureCount=2):
+    """
+    Trains a sklearn model (Specifically, logistic regression and Perceptron)
+    """
+    X = [df['f' + str(i)] for i in range(featureCount)]
+    Y = df['clicked']
+    model.fit(np.array(X).T, Y)
+    for i in range(featureCount):
+        model.coef_[0][i] = 1.0
+    if model.fit_intercept:
+        model.intercept_ = 1.0
+    else:
+        model.intercept_ = 0
```

Comment:
You could also try something in the lines of: 
```
model.intercept = float(model.fit_intercept)
```

Reply:
I'm trying here to set an initial value for the intercept to be 1.0 if the model is configured to use an intercept.

Reply:
I meant you could replace the whole if/else with a single statement 
```
model.intercept_ = float(model.fit_intercept)
```
Not too important though

Reply:
Why are you trying to set the intercept here?